tests/functional/openstackdataplanenodeset_controller_test.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package functional
+
+import (
+	"os"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	dataplanev1 "github.com/openstack-k8s-operators/dataplane-operator/api/v1beta1"
+	"github.com/openstack-k8s-operators/lib-common/modules/common/condition"
+	. "github.com/openstack-k8s-operators/lib-common/modules/common/test/helpers"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+var _ = Describe("Dataplane Role Test", func() {
+	var dataplaneNodeSetName types.NamespacedName
+	var dataplaneConfigMapName types.NamespacedName
+	var dataplaneSecretName types.NamespacedName
+
+	BeforeEach(func() {
+		dataplaneNodeSetName = types.NamespacedName{
+			Name:      "edpm-compute-nodeset",
+			Namespace: namespace,
+		}
+		dataplaneConfigMapName = types.NamespacedName{
+			Namespace: namespace,
+			Name:      "dataplanenodeset-edpm-compute-nodeset",
+		}
+		dataplaneSecretName = types.NamespacedName{
+			Namespace: namespace,
+			Name:      "dataplane-ansible-ssh-private-key-secret",
+		}
+		err := os.Setenv("OPERATOR_SERVICES", "../../config/services")
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	When("A Dataplane resorce is created", func() {
+		BeforeEach(func() {
+			DeferCleanup(th.DeleteInstance, CreateDataplaneNodeSet(dataplaneNodeSetName, DefaultDataPlaneNoNodeSetSpec()))
+		})
+		It("should have the Spec fields initialized", func() {
+			dataplaneNodeSetInstance := GetDataplaneNodeSet(dataplaneNodeSetName)
+			Expect(dataplaneNodeSetInstance.Spec.DeployStrategy.Deploy).Should(BeFalse())
+		})
+
+		It("should have the Status fields initialized", func() {
+			dataplaneNodeSetInstance := GetDataplaneNodeSet(dataplaneNodeSetName)
+			Expect(dataplaneNodeSetInstance.Status.Deployed).Should(BeFalse())
+		})
+
+		It("should have input not ready and unknown Conditions initialized", func() {
+			th.ExpectCondition(
+				dataplaneNodeSetName,
+				ConditionGetterFunc(DataplaneConditionGetter),
+				condition.ReadyCondition,
+				corev1.ConditionFalse,
+			)
+			th.ExpectCondition(
+				dataplaneNodeSetName,
+				ConditionGetterFunc(DataplaneConditionGetter),
+				condition.InputReadyCondition,
+				corev1.ConditionFalse,
+			)
+			th.ExpectCondition(
+				dataplaneNodeSetName,
+				ConditionGetterFunc(DataplaneConditionGetter),
+				dataplanev1.SetupReadyCondition,
+				corev1.ConditionFalse,
+			)
+
+			for _, cond := range []condition.Type{
+				condition.DeploymentReadyCondition,
```

Comment:
We will want to extend this to include all of the Baremetal status conditions, but at the moment I'm setting `PreProvisioned: true` and they aren't being initialized. Let's expand on this in a followup PR though.

Reply:
I'll just fix this and make it "When we use PreProvisioned nodes" It should .....

Then we can implement the Baremetal stuff in a seperate test scope.